sys/sys/mbuf.h

```diff
@@ -419,7 +419,8 @@ struct mbuf {
 				char		m_pktdat[0];
 			};
 		};
-		char	m_dat[0] __no_subobject_bounds;	/* !M_PKTHDR, !M_EXT */
+		/* !M_PKTHDR, !M_EXT */
+		char	m_dat[0] __subobject_use_remaining_size;
```

Comment:
Ah, I think we certainly need a fair bit more explanation in the log of what we are depending on the compiler doing for `m_dat` and `m_pktdat`.

Reply:
It occurs to me that I'm not sure whether the annotation is even necessary at this point. It could be a remnant of older compiler versions that did not know how to detect variable-sized arrays in certain situations. I need to check that, however I agree on the need for more context, I'll split these things into separate commits.

Reply:
It looks like this annotation is still required, while m_pktdat does not require the annotation. This is quite strange and possibly a compiler bug. I kept this in a separate commit, hopefully I can drop it if I find out what is going on in LLVM.

Reply:
Yeah definitely sounds like a compiler bug. Could you file a LLVM issue?